cassandra/deserializers.pyx

```diff
@@ -127,8 +127,8 @@ cdef class DesCounterColumnType(DesLongType):
 
 cdef class DesDateType(Deserializer):
     cdef deserialize(self, Buffer *buf, int protocol_version):
-        cdef double timestamp = unpack_num[int64_t](buf) / 1000.0
```

Comment:
There is a difference between the Python float data type and the C double.  We have more precision with the Python implementation (for large numbers). 

Python
```
In [215]: 2177403010.123456 * 1000000
Out[215]: 2177403010123456.0
```

C
```
cdef double test = 2177403010.123456
print test * 1000000
>>> 2.17740301012e+15 (217740301012000)
```

That's why I've decided to use int64 over a floating value for our extension. If you want, I can dig more in the Python float versus C double to give you the exact explanation and details about how many bits are for the precision etc.